Assets/Scripts/DrawLine.cs

```diff
@@ -60,11 +78,14 @@ public void PopulateWaypoints(List<Transform> wp)
                 waypoints.Add(wp[i].position);
             }
 
-            lineRenderer.positionCount = 1;
+            lineRenderer.positionCount = 1;             //setting the first waypoint immediately once we know where the waypoints are
             lineRenderer.SetPosition(0, waypoints[0]);
+            lineRenderer.positionCount++;
```

Comment:
Here you are setting the line renderers first point to the first waypoint which is right. But we also need to increment positionCount because NextWaypoint isn't called until the SECOND waypoint is hit by the player.  So we need to offset this by +1 everywhere else.

The reason for this is the first line segment is drawn between waypoints 0 and 1 so we need two position counts!